src/FSharpVSPowerTools.Logic/SyntaxConstructClassifier.fs

```diff
@@ -174,38 +173,45 @@ type SyntaxConstructClassifier
             None
         else Some()
 
+
     let mergeSpans (oldSpans: CategorizedColumnSpan<_>[]) (newSpans: CategorizedColumnSpan<_>[]) =
+
         let getLineRange spans =
             let lines = 
                 spans 
-                |> Array.filter (fun x -> 
+                |> Array.filterMap (fun x -> 
                     match x.Category with
                     | Category.Escaped 
                     | Category.Operator
                     | Category.Other
                     | Category.Printf
                     | Category.Quotation -> false
-                    | _ -> true) 
-                |> Array.map (fun x -> x.WordSpan.Line)
+                    | _ -> true)  (fun x -> x.WordSpan.Line)
+            if lines.Length = 0 then (-1, -1) else
             Array.min lines, Array.max lines
-            
-        let newStartLine, newEndLine = getLineRange newSpans
-        let oldStartLine, oldEndLine = getLineRange oldSpans
-        if newStartLine <= oldStartLine && newEndLine >= oldEndLine then
-            debug "[SyntaxConstructClassifier] Replace spans entirely."
-            newSpans
-        else
-            debug "[SyntaxConstructClassifier] Mergin spans (new range %A < old range %A)."
-                  (newStartLine, newEndLine) (oldStartLine, oldEndLine)
-
-            let oldSpans = 
-                seq { yield! oldSpans |> Seq.takeWhile (fun x -> x.WordSpan.Line < newStartLine)
-                      yield! oldSpans |> Seq.skipWhile (fun x -> x.WordSpan.Line < newEndLine) }
-
-            oldSpans
-            |> Seq.append newSpans
-            |> Seq.sortBy (fun x -> x.WordSpan.Line)
-            |> Seq.toArray
+
+        let compareSpans oldSpans newSpans =
+            let newStartLine, newEndLine = getLineRange newSpans
+            let oldStartLine, oldEndLine = getLineRange oldSpans
+            if newStartLine <= oldStartLine && newEndLine >= oldEndLine then
+                debug "[SyntaxConstructClassifier] Replace spans entirely."
+                newSpans
+            else
+                debug "[SyntaxConstructClassifier] Mergin spans (new range %A < old range %A)."
+                      (newStartLine, newEndLine) (oldStartLine, oldEndLine)
+                seq {
+                    yield (oldSpans |> Array.takeWhile (fun x -> x.WordSpan.Line < newStartLine))
+                    yield (oldSpans |> Array.skipWhile (fun x -> x.WordSpan.Line < newEndLine)) 
+                    yield newSpans
+                }   
+                |> Array.concat 
+                |> Array.sortBy (fun x -> x.WordSpan.Line)
```

Comment:
I don't think arrays are more preferable here. The old implementation does not allocate additional arrays.